mainBoard.py

```diff
@@ -0,0 +1,295 @@
+        self.shiprect = self.shiprect.move(self.ships[0].pos)
+        self.shipImages.append(self.shipImage.copy())
+        self.shiprects.append(self.shiprect)
+        # self.makeShip(tpos=[1000, 1000],tspeed=[10, 0])
+        # self.makeShip(tpos=[500, 500],tspeed=[0, 10])
+        # self.makeShip(tpos=[300, 100],tspeed=[10, 0])
+        # self.makeShip(tspeed=[50, 0])
+        self.make_dot()
+        """this is the main loop for the game"""
+        while 1:
+            """detection of escape key to close program"""
+            for event in pygame.event.get():
+                if event.type == pygame.QUIT:
+                    return
+                elif event.type == pygame.KEYDOWN and event.key == pygame.K_ESCAPE:
+                    pygame.display.quit()
+                    pygame.quit()
+                    sys.exit()
+            """this detects the arrow keys for turning and space for accelerations"""
+            keys = pygame.key.get_pressed()  #checking pressed keys
+            if keys[pygame.K_LEFT]:
+                self.rotShip(10, 0)
+            if keys[pygame.K_RIGHT]:
+                self.rotShip(-10, 0)
+            if keys[pygame.K_SPACE]:
+                self.accelerate(0)
+            if keys[pygame.K_r]:
+                board = BoardInit()
+
+            self.screen.fill(black)
+            i = 0
+            """below rotates ship to uppdate position"""
+            self.rotShip(0, i)
+            """this loops over each ship and does the necessry operations on each"""
+            while i < len(self.ships):
+                if i > 0:
+                    self.rotRock(2, i)
+                else:
+                    self.checkBounds(i)
+                self.moveShip(i)
+                self.gravity(i)
+                """blit will tie the bounding box and image to the screen"""
+                self.screen.blit(self.shipImages[i], self.shiprects[i])
+                self.check_dot()
+                self.checkHole(i)
+                i += 1
+
+            # pygame.draw.circle(self.screen, (255, 0, 0), (int(self.width/2), int(self.height/2)), 10, 0)
+            self.rotHole()
+            self.shipCollide()
+            self.rockCollide()
+            self.score_obj.render()
+            self.screen.blit(self.holeImage, self.holerect)
+            self.screen.blit(self.glowyImage, self.dotrect)
+            pygame.display.flip()
+            pygame.time.wait(33)
+        pygame.display.quit()
+        pygame.quit()
+        sys.exit()
+
+
+def rot_center(image, angle):
+    """rotate an image while keeping its center and size"""
+    orig_rect = image.get_rect()
+    rot_image = pygame.transform.rotate(image, angle)
+    rot_rect = orig_rect.copy()
+    rot_rect.center = rot_image.get_rect().center
+    rot_image = rot_image.subsurface(rot_rect).copy()
+    return rot_image
+
+
+if __name__ == '__main__':
+    board = BoardInit()
```

Comment:
It's a bit odd to start your game by initializing a class, but I guess it works.